src/Microsoft.Extensions.ML/Builder/PredictionEnginePoolBuilder.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.ML;
+
+namespace Microsoft.Extensions.ML
+{
+    /// <summary>
+    /// A class that provides the mechanisms to configure a pool
+    /// of ML.NET <see cref="PredictionEngine{TData, TPrediction}"/> objects.
+    /// </summary>
+    public class PredictionEnginePoolBuilder<TData, TPrediction>
+        where TData : class
+        where TPrediction : class, new()
+    {
+        /// <summary>
+        /// Initializes a new instance of <see cref="PredictionEnginePoolBuilder{TData, TPrediction}"/>.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> to add services to.</param>
+        public PredictionEnginePoolBuilder(IServiceCollection services)
+        {
+            Services = services ?? throw new ArgumentException(nameof(services));
+        }
+
+        /// <summary>
+        /// The <see cref="IServiceCollection"/> to add services to.
+        /// </summary>
+        public IServiceCollection Services { get; private set; }
```

Comment:
I am not sure what `IServiceCollection` is. Would each `Service` be used to create a `PredictionEnginePool`? Am the a targeted audience of this function (I don't have background of Azure function and ASP.NET)? Feel free to ignore this comment if I am not. :)

Reply:
> Am the a targeted audience of this function (I don't have background of Azure function and ASP.NET)?

Probably not. But when you start wanting to do more web service type of work, you will be :).

Check out https://docs.microsoft.com/en-us/aspnet/core/fundamentals/dependency-injection if you want to learn more about how the Dependency Injection system works in ASP.NET Core.